php/async/base/Exchange.php

```diff
@@ -0,0 +1,675 @@
+<?php
+
+/*
+
+MIT License
+
+Copyright (c) 2020 Paul Cook
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+*/
+
+//-----------------------------------------------------------------------------
+
+namespace ccxt_async;
+
+use Clue\React\Buzz\Browser;
+use \ccxt\NotSupported;
+use \ccxt\ExchangeNotAvailable;
+use \ccxt\ExchangeError;
+use \ccxt\DDoSProtection;
+use \ccxt\RequestTimeout;
+use Generator;
+use Psr\Http\Message\ResponseInterface;
+use React\EventLoop\LoopInterface;
+use React\Socket\Connector;
+use Recoil\React\ReactKernel;
+use Recoil\Recoil;
+
+// handle CCXT namespace constants
+foreach (get_defined_constants(true)['user'] as $const => $val) {
+    if (strpos($const, 'ccxt\\') === 0) {
+        define(str_replace('ccxt\\', '', $const), $val);
+    }
+}
+
+$version = '1.28.94';
+
+class Exchange extends \ccxt\Exchange {
+
+    const VERSION = '1.28.94';
+
+    /** @var Browser $browser */
+    public $browser;
+    /** @var LoopInterface $loop */
+    public $loop;
+    /** @var ReactKernel $kernel */
+    public $kernel;
+
+    public function __construct(LoopInterface $loop, ReactKernel $kernel, $options = array())
```

Comment:
We'd want to keep all constructor arguments inside `$options`, so that constructor initialization looks the same across all languages, regardless of sync/async mode. If you agree, would you help move them inside `$options`? Your involvement is appreciated very much!

Reply:
Sure, done in latest commit